pkg/repo/ent_impl_casbin_test.go

```diff
@@ -0,0 +1,84 @@
+package repo
+
+import (
+	"testing"
+
+	"github.com/dchest/uniuri"
+	"github.com/labstack/gommon/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEntImplCasbinRule(t *testing.T) {
+	entImpl, err := newEntImpl(log.New("test"))
+	assert.NoError(t, err)
+
+	t.Run("* actions", func(t *testing.T) {
+		hashedToken := uniuri.New()
+		_, err = entImpl.enforcer.AddPolicy(
+			hashedToken,
+			"example.com",
+			"/api/v1/facts",
+			mergeAllowedHTTPMethods([]string{"*"}),
+			"allow",
+		)
+		assert.NoError(t, err)
+		err = entImpl.enforcer.LoadPolicy()
+		assert.NoError(t, err)
+
+		shouldPass, err := entImpl.enforcer.Enforce(hashedToken, "example.com", "/api/v1/facts", "POST")
+		assert.True(t, shouldPass)
+		assert.NoError(t, err)
+	})
+
+	t.Run("POST to /api/v1/facts", func(t *testing.T) {
+		hashedToken := uniuri.New()
+		_, err = entImpl.enforcer.AddPolicy(
+			hashedToken,
+			"example.com",
+			"/api/v1/facts",
+			mergeAllowedHTTPMethods([]string{"POST"}),
+			"allow",
+		)
+		assert.NoError(t, err)
+		err = entImpl.enforcer.LoadPolicy()
+		assert.NoError(t, err)
+
+		shouldPass, err := entImpl.enforcer.Enforce(hashedToken, "example.com", "/api/v1/facts", "POST")
+		assert.True(t, shouldPass)
+		assert.NoError(t, err)
+
+		shouldPass, err = entImpl.enforcer.Enforce(hashedToken, "example.com", "/api/v1/facts/:id", "GET")
+		assert.False(t, shouldPass)
+		assert.NoError(t, err)
+	})
+
+	t.Run("POST to /api/v1/facts/noscope", func(t *testing.T) {
```

Comment:
example of adding policy for "/api/v1/facts/noscope"

Reply:
@jperichon 

A detailed explanation of what's going on for the authorization framework casbin. This URL is just an example, I'm testing here is the flexibility of enabling "Policy -> Enforcement" of authorization.

The convention we defined here
- https://casbin.org/docs/en/rbac-with-domains
- https://github.com/casbin/casbin/blob/master/examples/rbac_with_domains_model.conf
- OPV's own model is defined here https://github.com/open-privacy/opv/blob/main/pkg/repo/repo_ent_impl.go#L31-L49

Definitions:
```
[request_definition]
r = sub, dom, obj, act

[policy_definition]
p = sub, dom, obj, act, eft
```


The authorization framework
```
sub (i.e, subject, is the **hashed token string**)
dom (i.e, domain, is 1:1 mapped to **OPV's domain**, and facts,scopes,and grant token live in the domain)
obj (i.e., object, correctly we directly use **URL path** as the object for authorization)
act (i.e., action, OPV correctly map HTTP methods as the actions. In policy definition, it's defined as oneof `*`, `GET`, `POST`, `PUT`, and etc. And later merged as a regex string like `(GET)|(POST)`)
eft (i.e. effect, usually it's "allow", and we can also specify "deny" to have denylist style of policies)
```


Therefore, for example, to enable publishable tokens, we need to have a policy for each new grant token created. 

```
p, hashed_token_value_123, example.com, /api/v1/facts/noscope, POST, allow
```

The advance usage can be defined as a group role `g` `publishable_token`, and add tokens to this group.
```
p, publishable_token, example.com, /api/v1/facts/noscope, POST, allow


g, publishable_token, hashed_token_value_123, example.com
g, publishable_token, hashed_token_value_456, example.com
# We can add more tokens to this group publishable_token
```